File: zendrive-api/src/main/java/com/zendrive/api/core/service/role/RoleService.java
package com.zendrive.api.core.service.role;

import com.zendrive.api.core.model.dao.pgdb.auth.Role;
import com.zendrive.api.core.repository.zendrive.pgdb.RoleRepository;
import com.zendrive.api.exception.InvalidArgumentsException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class RoleService {
	private final RoleRepository roleRepository;

	public List<Role> getAll() {
		return roleRepository.findAll();
	}

	public Role getById(String id) {
		return roleRepository.findById(id)
												 .orElseThrow(() -> new InvalidArgumentsException("Role with provided id does not exist!"));
	}
}

File: zendrive-ui/src/api/services/RoleService.ts
import { AxiosResponse } from "axios";
import { authorizedAxiosApp } from "./Axios";
import { Role } from "@apiModels/auth/Role";

export class RoleService {
	static async getAll(): Promise<AxiosResponse<Role[]>> {
		return authorizedAxiosApp.get(`/role`);
	}

	static async getById(id: string): Promise<AxiosResponse<MetaFile>> {
		return authorizedAxiosApp.get(`/role/${id}`);
	}
}
